bokeh/io.py

```diff
@@ -516,7 +535,7 @@ def push_notebook(document=None, state=None, handle=None):
         msg = Document._compute_patch_between_json(handle.json, to_json)
     handle.comms.send(json.dumps(msg))
     handle._doc = document
-    handle._json = to_json
+    handle._json[id(document)] = to_json
```

Comment:
This now leaks private member (this was an issue before) and implementation detail (`id(...)`).